homeassistant/components/yardian/coordinator.py

```diff
@@ -0,0 +1,91 @@
+"""Update coordinators for Yardian."""
+
+from __future__ import annotations
+
+import datetime
+import logging
+from dataclasses import dataclass
+
+import async_timeout
+from pyyardian import AsyncYardianClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_WATERING_DURATION, DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+
+
+@dataclass
+class YardianDeviceState:
+    """Data retrieved from a Yardian device."""
+
+    zones: list[list]
+    active_zones: set[int]
+
+
+class YardianUpdateCoordinator(DataUpdateCoordinator[YardianDeviceState]):
+    """Coordinator for Yardian API calls."""
+
+    def __init__(
```

Comment:
Are there any API docs available for this device? Its great that its local, but it would be nicer if it was push

Reply:
Sorry we currently don't have any API docs for it because the local communication is also a new feature for our device. I'll discuss with my coworker to see if pushing is doable or not. But it'll be alright if we decide to just use the polling strategy for now right?

Reply:
It fine to use polling if push isn't possible, but its usually better to do it up front since it makes it much more work to switch later when you have to support older devices.